### ISSUE TITLE
false-positive `call-top-callable` after `callable()` narrowing on `int | Callable[..., int]`

### ISSUE THREAD

**Issue by sunny-zuo** [astral-sh/ty#2963]
### Summary

On `0.0.20`, [this code](https://play.ty.dev/eaa60f7b-b843-4083-9850-0fe41b1fb7b1) produces a diagnostic:

```python
from collections.abc import Callable

def f(x: int | Callable[[], int]) -> int:
    if callable(x):
        return x()  # Object of type `Top[(...) -> object]` is not safe to call; its signature is not known (call-top-callable)
    return x
```

This passes on `0.0.17`. This seems like its possibly a regression after being fixed in https://github.com/astral-sh/ruff/pull/22145?

### Version

ty 0.0.20

**Comment by MeGaGiGaGon** [comment]
This looks like a combination of #1578 plus #2221.

The #1578 part is straightforward, someone could subclass `int` and give it a `__call__`, resulting in a runtime error, so ty should be giving an error here:
https://play.ty.dev/4933c716-b6c5-4bb3-b002-b2efdfd8da83
```py
from collections.abc import Callable

def f(x: int | Callable[[], int]) -> int:
    if callable(x):
        reveal_type(x)  # Revealed type: `(int & Top[(...) -> object]) | (() -> int)`
        return x()  # Object of type `Top[(...) -> object]` is not safe to call; its signature is not known
    return x

print(f(1))  # Works as expected

class Sneaky(int):
    def __call__(self, arg: int) -> int:
        return arg

print(f(Sneaky(1)))  # Fails at runtime with TypeError: Sneaky.__call__() missing 1 required positional argument: 'arg'
```

The diagnostic here is not the best. Since it doesn't include the whole `(int & Top[(...) -> object]) | (() -> int)` part, unless you already knew about #1578 it's not clear why `Top[(...) -> object]` is the type and causing an issue. #2221 should already cover part of it, though I think it would be a good idea to put in the full type somewhere in the error instead of just `Top[(...) -> object]`.

The error itself does make sense, see https://docs.astral.sh/ty/reference/rules/#call-top-callable for an explanation.

**Comment by carljm** [comment]
For a concrete suggestion, using `if not isinstance(x, int):` instead of `if callable(x):` makes this code robust against a callable subtype of `int`, and makes ty happy with it.

Will close this as duplicate of #1578.
